feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -537,18 +580,264 @@ func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good, bad, lb []
+		for _, outPort := range expected_outgoing_port {
+			outgoing_traffic_counters := gnmi.OC().Interface(outPort.Name()).Counters()
+			outPkts := gnmi.Get(t, ate, outgoing_traffic_counters.InPkts().State())
+			receivedPkts = receivedPkts + outPkts
+		}
+
+		// Check if traffic restores with in expected time in milliseconds during modify NHG
+		if len(nonrx_ports) > 0 {
+			// Time took for traffic to restore in milliseconds after trigger
+			fpm := ((sentPkts - receivedPkts) / (fps / 1000))
+			if fpm > switchovertime {
+				t.Fatalf("Traffic loss %v msecs more than expected %v msecs", fpm, switchovertime)
+			}
+			t.Logf("Traffic loss during path change : %v msecs", fpm)
+		} else if sentPkts > receivedPkts {
+			t.Fatalf("Traffic didn't switch to the expected outgoing port")
 		}
 	}
 }
 
-func configStaticArp(p *ondatra.Port, ipv4addr string, macAddr string) *oc.Interface {
-	i := &oc.Interface{Name: ygot.String(p.Name())}
-	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+// configStaticArp configures static arp entries
+func configStaticArp(p string, ipv4addr string, macAddr string, trunk bool) *oc.Interface {
+	i := &oc.Interface{Name: ygot.String(p)}
+	if trunk {
+		i.Type = oc.IETFInterfaces_InterfaceType_ieee8023adLag
+	} else {
+		i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	}
 	s := i.GetOrCreateSubinterface(0)
 	s4 := s.GetOrCreateIpv4()
 	n4 := s4.GetOrCreateNeighbor(ipv4addr)
 	n4.LinkLayerAddress = ygot.String(macAddr)
 	return i
 }
+
+// TE3.7 case2 - Drain Implementation test.
+func testImplementDrain(ctx context.Context, t *testing.T, args *testArgs) {
+	configDUTDrain(t, args.dut)
+	addStaticRoute(t, args.dut)
+
+	t.Logf("Adding NHG %d, NH %d and NH %d  via gRIBI", nhg1ID, nh1ID, nh2ID)
+
+	if deviations.GRIBIMACOverrideWithStaticARP(args.dut) {
+		args.client.AddNH(t, nh1ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk2, Mac: port2mac, Dest: atePort2.IPv4})
+		args.client.AddNH(t, nh2ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk3, Mac: port3mac, Dest: atePort3.IPv4})
+
+	} else {
+		args.client.AddNH(t, nh1ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk2, Mac: port2mac})
+		args.client.AddNH(t, nh2ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk3, Mac: port3mac})
+
+	}
+	args.client.AddNHG(t, nhg1ID, map[uint64]uint64{nh1ID: 50, nh2ID: 50}, deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB)
+
+	t.Logf("Adding %d ipv4 vrf prefixes and gribi entries for them", ipv4Prefixcount)
+
+	prefixes := []string{}
+	for s := 0; s < ipv4Prefixcount; s++ {
+		prefixes = append(prefixes, generateIPAddress(dstPfxvalue, s))
+	}
+
+	for _, prefix := range prefixes {
```

Comment:
we are adding 700 prefixes in DEFAULT network-instance but only one prefix ( vip1 in L699) is used , any reason?

Reply:
Followed the readme procedure for the drain test, NHG#20 -> [ NH#20{ip: VIP1}, backupNH: NHG#10]. So one vip associated with NH20.